app/controllers/application_controller.rb

```diff
@@ -46,20 +66,31 @@ def n_plus_one_detection
   end
 
   # Its important that the location is NOT stored if:
-  # - The request method is not GET (non idempotent)
-  # - The request is handled by a Devise controller such as Devise::SessionsController as that could cause an
-  #    infinite redirect loop.
+  # - The request method is not GET (non idempotent).
+  # - The request is from Devise::SessionsController, could cause an infinite redirect loop.
   # - The request is an Ajax request as this can lead to very unexpected behaviour.
   # - The request is not a Turbo Frame request.
   def storable_location?
     request.get? &&
       is_navigational_format? &&
-      !devise_controller? &&
+      !authentication_controllers? &&
       !request.xhr? &&
       !turbo_frame_request?
   end
 
   def store_user_location!
     store_location_for(:user, request.fullpath)
   end
+
+  def after_sign_in_path_for(_)
```

Comment:
📝

This is overriden on a top-level and the argument is rejected because we'll always store the resource name as `:user` rather than `:email_authentication` or `:sso_authentication` for simplicity.